src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -259,6 +246,36 @@ private Map<Integer, Float> doANNSearch(final LeafReaderContext context, final i
             throw new RuntimeException(e);
         }
 
+        //From cardinality select different filterIds type
+        long[] filterIds;
+        FilterIdsSelectorType filterType;
+        if (filterIdsBitSet instanceof FixedBitSet) {
+            /**
+             * When filterIds is dense filter, using fixed bitset
+             */
+            filterIds = ((FixedBitSet) filterIdsBitSet).getBits();
+            filterType = FilterIdsSelectorType.BITMAP;
+        } else if(cardinality < MAX_ID_SELECT_ARRAY) {
```

Comment:
Can we move this logic https://github.com/opensearch-project/k-NN/blob/main/jni/src/faiss_wrapper.cpp#L484 to here including the comments. 

You can also remove the code. https://github.com/opensearch-project/k-NN/blob/main/jni/src/faiss_wrapper.cpp#L484

Reply:
Fixed at https://github.com/opensearch-project/k-NN/pull/1402/commits/bfbfb55ab8eaedee1e19894ab5001e3339e9c00a